analyses/chromosomal-instability/02a-plot-chr-instability-heatmaps.Rmd

```diff
@@ -0,0 +1,300 @@
+    col = list(histologies = histologies_colors),
+    which = "row",
+    show_annotation_name = FALSE
+  )
+  # Plot on a heatmap
+  heatmap <- ComplexHeatmap::Heatmap(binned_counts_mat,
+    col = col_fun,
+    heatmap_legend_param = list(title = "Count of chr breaks"),
+    cluster_columns = FALSE,
+    cluster_rows = FALSE,
+    show_column_names = FALSE,
+    show_row_names = FALSE,
+    bottom_annotation = chr_annot,
+    left_annotation = hist_annot,
+    na_col = "#f1f1f1"
+  )
+  # Return plot
+  return(heatmap)
+}
+```
+
+### Read in data 
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(metadata_file)
+```
+
+Load in the previously localized breakpoint data. 
+
+```{r}
+# Read in each dataset
+binned_counts <- lapply(binned_counts_files, readr::read_tsv)
+
+# Name them
+names(binned_counts) <- gsub(
+  "breakpoint-data/|_binned_counts.tsv",
+  "",
+  binned_counts_files
+)
+```
+
+Extract chromosome labels and make an alternating color key for them. 
+
+```{r}
+# Extract chromosome labels
+chrs <- paste0("chr", binned_counts[[1]]$chr_bin_names)
+
+# Make chromosome labeling `HeatmapAnnotation` object.
+chrs <- as.factor(chrs)
+
+# Make a key for assigning alternating colors to the chromosomes
+chr_colors <- rep(c("grey", "lightblue"), length.out = length(unique(chrs)))
+names(chr_colors) <- unique(chrs)
+
+# Make the same order as the data.frame
+chr_colors <- as.character(dplyr::recode(chrs, !!!chr_colors))
+names(chr_colors) <- chrs
+```
+
+Get sample names. 
+
+```{r}
+common_samples <- grep("chr_bin_names",
+  colnames(binned_counts[[1]]),
+  invert = TRUE,
+  value = TRUE
+)
+```
+
+### Set up for making heatmaps of the breakpoints
+
+Given the `GenomicRanges` objects for each sample, create a combined plot for 
+each. 
+
+Make histology labeling `HeatmapAnnotation` object.
+
+```{r}
+# Get the histologies for the samples in this set and order them by histology
+histologies <-
+  data.frame(Kids_First_Biospecimen_ID = common_samples) %>%
+  dplyr::inner_join(dplyr::select(metadata, Kids_First_Biospecimen_ID, short_histology)) %>%
+  dplyr::mutate(short_histology = as.factor(short_histology)) %>%
+  dplyr::filter(!is.na(short_histology)) %>%
+  dplyr::arrange(short_histology) %>%
+  tibble::column_to_rownames("Kids_First_Biospecimen_ID")
+
+# Get values that can be used to make colors equi distant hues away for the
+# number of histology groups we have
+col_val <- seq(
+  from = 0, to = 1,
+  length.out = length(unique(histologies$short_histology))
+)
+
+# Translate into colors
+histologies_colors <- hsv(h = col_val, s = col_val, v = 1)
+
+# Make this named based on histology
+names(histologies_colors) <- unique(histologies$short_histology)
+
+# Make the same order as the data.frame
+histologies_colors <- as.character(dplyr::recode(histologies$short_histology, !!!histologies_colors))
+
+# Make the names
+names(histologies_colors) <- as.character(rownames(histologies))
```

Comment:
After fixing the naming in complexheatmap function, these steps are not needed. The color vector only needs one named value for each category. Same for the `chr_colors` steps on lines  65-67.